Ginger/GingerCoreNET/ActionsLib/ActCliOrchestration.cs

```diff
@@ -0,0 +1,194 @@
+        {
+            public static string SourceFilePath = "SourceFilePath";
+            public static string FileOperationMode = "FileOperationMode";
+            public static string DestinationFolder = "DestinationFolder";
+        }
+
+        [IsSerializedForLocalRepository]
+        public string ScriptInterpreter { get; set; }
+
+        [IsSerializedForLocalRepository]
+        public string ScriptPath { get; set; }
+
+        [IsSerializedForLocalRepository]
+        public bool WaitForProcess { get; set; }
+
+        [IsSerializedForLocalRepository]
+        public bool ParseResult { get; set; }
+
+        [IsSerializedForLocalRepository]
+        public string Delimiter { get; set; }
+
+        String DataBuffer;
+        public override async void Execute()
+        {
+            DataBuffer = string.Empty;
+            if (ParseResult && string.IsNullOrEmpty(ValueExpression.Calculate(Delimiter)))
+            {
+                Error = "Delimiter is Empty";
+                return;
+            }
+
+            if(WaitForProcess)
+            {
+                await ExecuteCliProcess();
+            }
+            else
+            {
+               ExecuteCliProcess();
+            }
+        }
+
+        private async Task ExecuteCliProcess()
+        {
+            StringBuilder argumentsstring = new StringBuilder();
+            foreach (var p in this.InputValues)
+            {
+                argumentsstring.Append(p.Param).Append(" ");
+                if (!string.IsNullOrEmpty(p.ValueForDriver))
+                {
+                    argumentsstring.Append(p.ValueForDriver).Append(" ");
+                }
+            }
+
+            if (!string.IsNullOrEmpty(this.ScriptInterpreter))
+            {
+                string path = String.Empty;
+                Status = Amdocs.Ginger.CoreNET.Execution.eRunStatus.Pending;
+                if (WorkSpace.Instance.Solution.LoggerConfigurations.CalculatedLoggerFolder != null)
+                {
+                    string folderPath = $"{WorkSpace.Instance.Solution.Folder}{Path.DirectorySeparatorChar}Documents{Path.DirectorySeparatorChar}CLIOrchestration";
+                    if (!Directory.Exists(folderPath))
+                    {
+                        Directory.CreateDirectory(folderPath);
+                    }
+                    string DatetimeFormate = DateTime.Now.ToString("ddMMyyyy_HHmmssfff");
+                    string Filename = $"{ItemName}_{DatetimeFormate}.txt";
+                    path = $"{folderPath}{Path.DirectorySeparatorChar}{Filename}";
+                    AddOrUpdateReturnParamActual(ParamName: "CLIOrchestration LogFile", ActualValue: path);
+                }
+
+                var cmd = Cli.Wrap(this.ScriptInterpreter)
+                                .WithArguments(argumentsstring.ToString()) | (PipeTarget.ToDelegate(parseRcwithDelimiter));
+                try
+                {
+                    var Result = await cmd.ExecuteAsync();
```

Comment:
It should not wait for results and action status in case of WaitForProcess is false, also output log file should not be generated